configure.py

```diff
@@ -1177,16 +1180,30 @@ def remove_dups(parts):
     Form snippets of makefile for building each source file
     """
     def build_commands(sources, obj_dir, flags):
```

Comment:
The duplication of the entire yield expression here bothers me. Could this be written using a %s format that consumes `(cc.add_include_dir_option if options.with_external_includedir else '')`